modules/drivers/bigquery-cloud-sdk/src/metabase/driver/bigquery_cloud_sdk.clj

```diff
@@ -65,25 +65,23 @@
 
 (defn- list-tables
   "Fetch all tables (new pages are loaded automatically by the API)."
-  (^Iterable [database]
-   (list-tables database false))
-  (^Iterable [{{:keys [project-id dataset-filters-type dataset-filters-patterns]} :details, :as database} validate-dataset?]
-   (list-tables (database->client database)
-                (or project-id (bigquery.common/database-details->credential-project-id (:details database)))
-                dataset-filters-type
-                dataset-filters-patterns
-                (boolean validate-dataset?)))
-  (^Iterable [^BigQuery client ^String project-id ^String filter-type ^String filter-patterns ^Boolean validate-dataset?]
-   (let [datasets (.listDatasets client project-id (u/varargs BigQuery$DatasetListOption))
-         inclusion-patterns (when (= "inclusion" filter-type) filter-patterns)
-         exclusion-patterns (when (= "exclusion" filter-type) filter-patterns)
+  (^Iterable [database-details]
+   (list-tables database-details {:validate-dataset? false}))
+  (^Iterable [{:keys [project-id dataset-filters-type dataset-filters-patterns] :as details} {:keys [validate-dataset?]}]
+   (let [client (database-details->client details)
+         project-id (or project-id (bigquery.common/database-details->credential-project-id details))
+         datasets (.listDatasets client project-id (u/varargs BigQuery$DatasetListOption))
+         inclusion-patterns (when (= "inclusion" dataset-filters-type) dataset-filters-patterns)
+         exclusion-patterns (when (= "exclusion" dataset-filters-type) dataset-filters-patterns)
```

Comment:
This is just a refactoring. The old 2-arity version was unnecessary and confusing because it changed the names of the args